pkg/controller/daemon/controller.go

```diff
@@ -354,6 +377,15 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) {
 	var nodesNeedingDaemonPods, podsToDelete []string
 	for _, node := range nodeList.Items {
 		shouldRun := nodeShouldRunDaemonPod(&node, ds)
+		// If the node is not ready, don't run on it.
+		// TODO(mikedanese): remove this once daemonpods forgive nodes
+		shouldRun = shouldRun && api.IsNodeReady(&node)
+
+		// If the node is unschedulable, don't run it
+		// TODO(mikedanese): remove this once we have the right node admitance levels.
+		// See https://github.com/kubernetes/kubernetes/issues/17297#issuecomment-156857375.
+		shouldRun = shouldRun && !node.Spec.Unschedulable
+
```

Comment:
but the nodeShouldRunDaemonPod function is also used by other functions such as addNode, UpdateNode and updateDaemonSetStatus, in those functions, they don't need to check node.Spec.Unschedulable and api.IsNodeReady(&node). This is the reason I remove this two lines from nodeShouldRunDaemonPod function. Thanks